mist-api/src/routes/genes/get.js

```diff
@@ -1,17 +1,41 @@
 'use strict'
 
+// Local
+const searchUtil = require('lib/util')
+
 module.exports = function(app, middlewares, routeMiddlewares) {
 	let models = app.get('models'),
 		helper = app.get('lib').RouteHelper.for(models.Gene)
 
+	const exactMatchFieldNames = [
+			'version',
+			'locus',
+			'old_locus',
+			'stable_id'
+	]
+	const textFieldNames = ['product']
+
 	return [
 		middlewares.parseCriteriaForMany(models.Gene, {
 			accessibleModels: [
+				models.Genome,
 				models.Component,
 				models.Aseq,
 				models.Dseq
-			]
+			],
+			permittedWhereFields: [
+				'id'
+			],
 		}),
+		(req, res, next) => {
+			// Provide for searching against textFieldNames (see above)
+			if (Reflect.has(req.query, 'search')) {
+				searchUtil.assignExactMatchCriteria(req.query.search, res.locals, exactMatchFieldNames)
+				searchUtil.assignInexactMatchCriteria(req.query.search, res.locals, textFieldNames)
```

Comment:
Fine for now, but we will probably want to make this into a middleware. Have you checked out the `parse-query-params` middleware? I'm wondering if we can somehow wrap it to:
a) perform the queryValue trimming (via the `transform` argument) 
b) update the criteria